osu.Game.Rulesets.Catch/Objects/Drawables/DrawableFruit.cs

```diff
@@ -11,18 +12,48 @@ namespace osu.Game.Rulesets.Catch.Objects.Drawables
 {
     public class DrawableFruit : DrawablePalpableCatchHitObject
     {
+        public readonly Bindable<FruitVisualRepresentation> VisualRepresentation = new Bindable<FruitVisualRepresentation>();
+
+        protected virtual FruitVisualRepresentation GetVisualRepresentation(int indexInBeatmap) => (FruitVisualRepresentation)(indexInBeatmap % 4);
+
+        private FruitPiece fruitPiece;
+
         public DrawableFruit(CatchHitObject h)
             : base(h)
         {
         }
 
         [BackgroundDependencyLoader]
         private void load()
+        {
+            ScaleContainer.Rotation = (float)(RNG.NextDouble() - 0.5f) * 40;
+
+            IndexInBeatmap.BindValueChanged(change =>
+            {
+                VisualRepresentation.Value = GetVisualRepresentation(change.NewValue);
+            }, true);
+
+            VisualRepresentation.BindValueChanged(_ => updatePiece());
+            HyperDash.BindValueChanged(_ => updatePiece(), true);
+        }
+
+        private void updatePiece()
         {
             ScaleContainer.Child = new SkinnableDrawable(
-                new CatchSkinComponent(getComponent(HitObject.VisualRepresentation)), _ => new FruitPiece());
+                new CatchSkinComponent(getComponent(VisualRepresentation.Value)),
+                _ => fruitPiece = new FruitPiece
+                {
+                    VisualRepresentation = { BindTarget = VisualRepresentation },
+                    HyperDash = { BindTarget = HyperDash },
+                });
+        }
 
-            ScaleContainer.Rotation = (float)(RNG.NextDouble() - 0.5f) * 40;
+        protected override void Update()
+        {
+            base.Update();
+
+            if (fruitPiece != null)
+                fruitPiece.Border.Alpha = (float)Math.Clamp((StartTimeBindable.Value - Time.Current) / 500, 0, 1);
```

Comment:
I don't like skinnable members' properties being set from the parent DHO. Would prefer moving this back into `FruitPiece` and instead of `hitObject = drawableObject.HitObject` as it originally did, it could directly do `border.Alpha = ... drawable.HitObject.StartTime - Time.Current ...`

Unless there's a bigger reason for doing it this way?

Reply:
Yes, one reason is to make pieces usable from non-DHO fruits on the plate.
Another reason having a `HitObject` on `FruitPiece` is not good (I think) is it has to react when the hit object is changed (because it will be recycled through pooling), but because it is not a DHO, `OnAppy` and `OnFree` doesn't exist and we have to manually make a method similar to it.
Then, another way to implement the logic is having a `Bindable` for `StartTime`. But I think the current way works better for the fruits on the plate.

Reply:
On what level will pooling happen for catch? Is it going to be:
1. A piece pool which _both_ DHOs and the plated objects use, or:
2. A piece pool for just the plated objects, and the DHOs being assigned pieces statically at construction time?
If the second, pieces can DI the DHO, which would be `null` for plated ones.

Reply:
I'm thinking of the option 2.
Oh, so the suggestion is that storing DHO, not HO in a `FruitPiece`, right? (so the hit object change is not needed as I described, as it will be accessed through the DHO each update).
I think it works.

Reply:
Yep, that's exactly it. Already done that way for other skinnable pieces osu!-side.